hydroflow/src/compiled/pull.rs

```diff
@@ -11,6 +11,82 @@ where
 {
     state: L::Repr,
 }
+pub struct StreamJoinState<K, L>
+where
+    L: LatticeRepr,
+{
+    tab: HashMap<K, L::Repr>,
+}
+
+impl<K, L> Default for StreamJoinState<K, L>
+where
+    L: LatticeRepr,
+{
+    fn default() -> Self {
+        Self {
+            tab: HashMap::new(),
+        }
+    }
+}
+
+pub struct StreamJoin<'a, K, Buf, L, Update, Stream, StreamV>
+where
+    K: Eq + std::hash::Hash,
+    L: LatticeRepr + Merge<Update>,
+    Update: LatticeRepr,
+    Buf: Iterator<Item = (K, Update::Repr)>,
+    Stream: Iterator<Item = (K, StreamV)>,
+{
+    buf: Buf,
+    stream: Stream,
+    state: &'a mut StreamJoinState<K, L>,
+    _marker: PhantomData<Update>,
+}
+
+impl<'a, K, Buf, L, Update, Stream, StreamV> Iterator
+    for StreamJoin<'a, K, Buf, L, Update, Stream, StreamV>
+where
+    K: Eq + std::hash::Hash + Clone,
+    L: LatticeRepr + Merge<Update>,
+    L::Repr: Default,
+    Update: LatticeRepr,
+    StreamV: Clone,
+    Buf: Iterator<Item = (K, Update::Repr)>,
+    Stream: Iterator<Item = (K, StreamV)>,
+{
+    type Item = (K, StreamV, L::Repr);
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
Again kind of curious why this isn't more explicitly similar to SymmetricHashJoin